### PR TITLE
SkyHigh Security integration

### PR DESCRIPTION
**Pull request by MosheEichler**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-1964

## Description
Added a new pack with an integration to McAfee MVISION

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_test.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import demistomock as demisto
+from McAfeeMVisionCASB import main
+
+
+def util_load_json(path):
+    with io.open(path, mode='r', encoding='utf-8') as f:
+        return json.loads(f.read())
+
+
+def test_incident_query_command(mocker):
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_test.py:14]: add docstrings to all unit tests (when - given - then)

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml [hunk 2]
```diff
@@ -0,0 +1,361 @@
+category: CASB
+commonfields:
+  id: McAfee MVision CASB
+  version: -1
+configuration:
+- defaultvalue: https://www.myshn.net/
+  display: Your server URL
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml:7]:
```suggestion
  display: Base URL (e.g., https://www.myshn.net)
```

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    name: mvision-casb-anomaly-activity-list
+    outputs:
+    - contextPath: MVisionCASB.AnomalyActivity.timeStamp
+      description: The timestamp
+      type: String
+    - contextPath: MVisionCASB.AnomalyActivity.actionName
+      description: The action name
+      type: String
+    - contextPath: MVisionCASB.AnomalyActivity.asnName
+      description: ""
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml:175]: update all descriptions, try using [demisto-sdk generate-outputs --ai](https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/generate_outputs/README.md#generate-outputs)

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    base_url = urljoin(demisto.params()['url'].removesuffix('/'), '/shnapi/rest')
+    verify_certificate = not demisto.params().get('insecure', False)
+    credentials = demisto.params().get('credentials', {})
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py:291]:
```suggestion
    params = demisto.params()
    base_url = urljoin(params['url'].removesuffix('/'), '/shnapi/rest')
    verify_certificate = not params.get('insecure', False)
    credentials = params.get('credentials', {})
```

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+                'Time(UTC)': incident.get('timeCreated'),
+                'Status': incident.get('status'),
+                'Alert Action': incident.get('remediationResponse'),
+                'Service Name': incident.get('serviceNames'),
+                'Alert Severity': incident.get('incidentRiskSeverity'),
+                'User Name': incident.get('actorId'),
+                'Policy Name': incident.get('policyName'),
+            })
+
+        demisto.debug(f'This is the output: {readable_dict}')
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py:190]: ?

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    """
+
+    try:
+        client.test()
+        message = 'ok'
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.py:110]:
```suggestion
            raise
```

**Review by yaakovi — CHANGES_REQUESTED** [review]
very nice work

File: Packs/CommonTypes/ReleaseNotes/3_2_38.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#### Incident Fields
+- Added the following incident fields to be associated with the McAfee MVision CASB Alert and McAfee MVision CASB Threat incident types.
```
Comment (yaakovi) [inline Packs/CommonTypes/ReleaseNotes/3_2_38.md:2]:
```suggestion
- Added the following incident fields to be associated with the *McAfee MVision CASB Alert* and *McAfee MVision CASB Threat* incident types.
```

File: Packs/Malware/ReleaseNotes/1_4_2.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
+#### Incident Fields
+- **Malware Name**
```
Comment (yaakovi) [inline Packs/Malware/ReleaseNotes/1_4_2.md:3]: you didn't change the field

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_description.md [hunk 9]
```diff
@@ -1,8 +1,11 @@
...
-Markdown file for integration configuration  help snippet. In this file add:
-- Brief information about how to retrieve the API key of your product
-- Other useful information on how to configure your integration in XSOAR
 
-Since this is a Markdown file, we encourage you to use MD formatting for sections, sub-sections, lists, etc.
+### Base URL
+The base url for example https://www.myshn.net
 
+### credentials
+UserName and Password
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_description.md:8]: redundant

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_description.md [hunk 10]
```diff
@@ -1,8 +1,11 @@
-## BaseIntegration Help
+## McAfee MVision CASB
 
-Markdown file for integration configuration  help snippet. In this file add:
-- Brief information about how to retrieve the API key of your product
-- Other useful information on how to configure your integration in XSOAR
 
-Since this is a Markdown file, we encourage you to use MD formatting for sections, sub-sections, lists, etc.
+### Base URL
+The base url for example https://www.myshn.net
```
Comment (yaakovi) [inline Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB_description.md:5]: redundant

**Review by yaakovi — COMMENTED** [review]
nice work

**Comment by MosheEichler** [comment]
The pre isn't done yet because we are waiting for more credentials to implement the anomaly command.
[Jira](https://jira-hq.paloaltonetworks.local/browse/CRTX-52848) tiket

**Comment by ShirleyDenkberg** [comment]
@michalgold @altmannyarden @yaakovi @idovandijk Doc review completed.

File: Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml (hunk 3 above)
Comment (yaakovi) [reply Packs/McAfeeMVisionCASB/Integrations/McAfeeMVisionCASB/McAfeeMVisionCASB.yml]: yes

**Review by yaakovi — APPROVED** [review]
* merge the two mappers
* set the classifier and mapper as defaults in the integration yml

**Comment by MosheEichler** [comment]
@mchasepan
Hi. Yes the pr is ready will be merged after the build will succeed.

File: Packs/SkyhighSecurity/pack_metadata.json [hunk 11]
```diff
@@ -0,0 +1,17 @@
+{
+    "name": "Skyhigh Security",
```
Comment (mchasepan) [inline Packs/SkyhighSecurity/pack_metadata.json:2]:
```suggestion
    "name": "Skyhigh Security SSE",
```

File: Packs/SkyhighSecurity/pack_metadata.json [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+    "description": "Skyhigh Security is a cloud-based, multi-tenant service that enables Cloud Discovery and Risk Monitoring, Cloud Usage Analytics, Cloud Access and Control.\n",
```
Comment (mchasepan) [inline Packs/SkyhighSecurity/pack_metadata.json:3]:
```suggestion
    "description": "Skyhigh Security is a cloud-based, multi-tenant service that enables Cloud Discovery and Risk Monitoring, Cloud Usage Analytics, Cloud Access and Control.",
```

**Pull request merged**